File: DAA2/skipmulti.c
#include <stdio.h>
#include <string.h>

int main() {
    char s1[100], s2[100];
    printf("Enter two strings:\n");
    scanf("%s %s", s1, s2);

    int m = strlen(s1);
    int n = strlen(s2);
    int flag = -1;

    for (int i = 0; i < m; i++) {
        if (s1[i] == s2[0]) {
            int count = 0;
            int a = 0;
            for (int j = 0; j < n; j++) {
                if ((i + j + a) < m && s2[j] == s1[i + j + a]) {
                    count++;
                } else {
                    break;
                }
                a += j;
            }
            if (count == n) {
                flag = i;
                break;
            }
        }
    }

    if (flag != -1) {
        printf("Match found from index %d\n", flag);
    } else {
        printf("No match found\n");
    }

    return 0;
}

File: REVISION/Brute force/Skipmulti.java
import java.util.*;
class one{
    int count;
    int ind;
    one(String a, String b){
        count = 0;
        ind = -1;
        for(int i=0; i<a.length()-b.length(); i++){
            if(a.charAt(i) == b.charAt(0)){
                int f = 0;
                int skip = 1;
                int k = i;
                for(int j=0; j<b.length(); j++){
                    skip += 1;
                    if(b.charAt(j) != a.charAt(k)){
                        f = 1;
                        break;
                    }
                    k = k+skip;
                }
                if(f == 0){
                    ind = i;
                    count++;
                }
            }
        }
    }
}
public class Skipmulti{
    public static void main(String[] args){
        Scanner in = new Scanner(System.in);
        System.out.println("Enter two strings: ");
        String a = in.nextLine();
        String b = in.nextLine();
        one o = new one(a,b);
        if(o.count > 0){
            System.out.println("Match found at "+o.ind+"."); 
            System.out.println("Count is: "+o.count+".");
        }else{
            System.out.println("Match not found.");
        }
    }
}